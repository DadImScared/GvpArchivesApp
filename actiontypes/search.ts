export const UPDATE_SEARCH_QUERY = "search/UPDATE_SEARCH_QUERY";
export const TOGGLE_CATEGORIES = "search/TOGGLE_CATEGORIES";
export const SELECT_ALL_CATEGORIES = "search/SELECT_ALL_CATEGORIES";
export const REMOVE_ALL_CATEGORIES = "search/REMOVE_ALL_CATEGORIES";
