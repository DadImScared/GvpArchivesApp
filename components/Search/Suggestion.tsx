import * as React from "react";
import { StyleSheet } from "react-native";

import { Button, NativeBase, Text } from "native-base";

interface IProps extends NativeBase.Button {
  suggestion: string;
  onPressSuggestion: (suggestion: string) => void;
}

const styles: any = StyleSheet.create({
  button: { margin: 8, padding: 8, paddingTop: 8, paddingBottom: 8 }
});

export class Suggestion extends React.Component<IProps> {

  onPressSuggestion = () => {
    this.props.onPressSuggestion(this.props.suggestion);
  };

  render() {
    const { suggestion, onPressSuggestion, ...other } = this.props;
    return (
      <Button small={true} primary={true} style={styles.button} onPress={this.onPressSuggestion} {...other}>
        <Text>{suggestion}</Text>
      </Button>
    );
  }
}

export default Suggestion;
