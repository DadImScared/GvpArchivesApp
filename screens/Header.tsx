import * as React from "react";

import { Ionicons } from "@expo/vector-icons";
import { NavigationScreenProps } from "react-navigation";
import HeaderButtons, { HeaderButton } from "react-navigation-header-buttons";

const IoniconsHeaderButton = (props: any) => (
  <HeaderButton {...props} IconComponent={Ionicons} iconSize={23} color={"blue"} />
);

export const IoniconsHeaderButtons = (props: any) => (
  <HeaderButtons
    HeaderButtonComponent={IoniconsHeaderButton}
    {...props}
  />
);

export const HeaderItem = HeaderButtons.Item;

class HeaderRight extends React.Component<NavigationScreenProps> {

  searchPress = () => {
    this.props.navigation.navigate("SearchBar");
  };

  render() {
    return (
      <IoniconsHeaderButtons>
        <HeaderItem iconName={"ios-search"} onPress={this.searchPress} title={"Search"} />
      </IoniconsHeaderButtons>
    );
  }
}

export const headerRight = ({ navigation }: NavigationScreenProps) => (
  <HeaderRight navigation={navigation} />
);
