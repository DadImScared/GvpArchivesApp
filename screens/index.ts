export { default as AutoComplete } from "./SearchBarScreen/AutoComplete";
export { default as Filter } from "./SearchBarScreen/Filter";
export { default as SearchHeader } from "./SearchBarScreen/Header";
export { default as RecentSearches } from "./SearchBarScreen/RecentSearches";
export { default as ItemsByCategory } from "./ItemsByCategory";
export { default as Search } from "./Search";
