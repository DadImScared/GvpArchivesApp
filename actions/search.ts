import axios, { AxiosPromise } from "axios";
import { AnyAction, Dispatch } from "redux";
import { action } from "typesafe-actions";

import * as ActionTypes from "../actiontypes/search";
import { AUTOCOMPLETE_QUERY } from "../loadingPrefix";
import { IReducerState } from "../reducers";
import { getQueryId } from "../reducers/search";
import autoComplete from "./autoComplete";
import { BASE_URL } from "./itemsById";

const autoCompleteUrl = `${BASE_URL}/api/v1/completeme/`;

export const isValidValidTimeToLive = (time: number) => time ? time >= +new Date() : false;

const updateSearchQuery = (query: string) => action(ActionTypes.UPDATE_SEARCH_QUERY, query);

type ApiCallCallback<ReturnType> = (
  state: IReducerState,
  dispatch: Dispatch
) => ReturnType;

type IAfterApiCall<Data> = (
  data: Data,
  state: IReducerState,
  dispatch: Dispatch
) => Promise<AnyAction[]> | AnyAction[];

export interface IApiCallConfig<Data> {
  afterApiCall?: IAfterApiCall<Data>;
  apiCall: ApiCallCallback<AxiosPromise<any>>;
  beforeApiCall?: ApiCallCallback<Promise<AnyAction[]> | AnyAction[]>;
  loadingId?: ((state: IReducerState) => string) | string;
  shouldCallApi?: ApiCallCallback<boolean | AxiosPromise<void>>;
}

export const getQuerySuggestions = (query: string, categories: string[]): IApiCallConfig<string[]> => {
  const queryId = getQueryId(query, categories);
  const loadingId = `${AUTOCOMPLETE_QUERY}${queryId}`;
  return {
    loadingId,
    afterApiCall: (data) => {
      return [
        autoComplete.addAutoComplete(
          query,
          categories,
          data.map((textObj: any) => textObj.text)
        )
      ];
    },
    apiCall: () => axios.get(`${autoCompleteUrl}${query}`),
    shouldCallApi: (state) => {
      const isLoading = state.loading[loadingId];
      if (!query || isLoading) return false;

      const currentAutoComplete = state.autoComplete[queryId];
      if (!currentAutoComplete) return true;

      return isValidValidTimeToLive(currentAutoComplete.timeToLive) ?
        false
        :
        axios.get(`${autoCompleteUrl}${query}`);
    }
  };
};

export default {
  updateSearchQuery
};
