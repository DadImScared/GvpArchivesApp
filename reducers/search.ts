import _ from "lodash";
import { ActionType } from "typesafe-actions";

import { search } from "../actions";
import * as ActionTypes from "../actiontypes/search";
import { IReducerState } from "./index";

export interface ISearchState {
  categories: string[];
  query: string;
}

export type SearchStateAndActions<
  T extends keyof ISearchState | keyof typeof search
  > = Pick<ISearchState & typeof search, T>;

export const CATEGORIES = [
  "bhagavatpatrika",
  "book",
  "harikatha",
  "harmonistmagazine",
  "harmonistmonthly",
  "lecture",
  "movie",
  "song"
];

export const getInitialSearchState = (): ISearchState => ({
  categories: CATEGORIES,
  query: ""
});

export default function(state = getInitialSearchState(), action: ActionType<typeof search>) {
  switch (action.type) {
    case ActionTypes.UPDATE_SEARCH_QUERY:
      return {
        ...state,
        query: action.payload
      };
    case ActionTypes.TOGGLE_CATEGORIES:
      const { payload } = action;
      const { categories } = state;
      return {
        ...state,
        categories: categories.includes(payload) ?
          categories.filter((category) => category !== action.payload)
          :
          [...categories, payload]
      };
    case ActionTypes.SELECT_ALL_CATEGORIES:
      return {
        ...state,
        categories: [...CATEGORIES]
      };
    case ActionTypes.REMOVE_ALL_CATEGORIES:
      return {
        ...state,
        categories: []
      };
    default:
      return state;
  }
}

export const getQueryId = (query: string, categories: string[]) => {
  return `${query}-${_.sortBy(categories, (o) => o).join(",")}`;
};

export const getSearch = (state: IReducerState) => state.search;
