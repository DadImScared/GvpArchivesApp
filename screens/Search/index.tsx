import * as React from "react";
import { FlatList, View } from "react-native";
import { NavigationScreenProps } from "react-navigation";
import { connect } from "react-redux";
import { compose, hoistStatics, shouldUpdate } from "recompose";

import { Text } from "native-base";

import { recentSearches } from "../../actions";
import { getSearchResults } from "../../actions/searchResults";
import Loading from "../../components/Loading";
import RenderItem from "../../components/RenderItem";
import { Suggestion } from "../../components/Search";
import { IReducerState } from "../../reducers";
import { getQueryId } from "../../reducers/search";
import { ISearchResultData, makeSelectSearchResults } from "../../reducers/searchResults";
import { getNavigationParams } from "../../utils/navigation";
import { headerRight } from "../Header";

interface IProps extends NavigationScreenProps, ISearchResultData {
  addRecentSearch: (query: string, categories: string[]) => void;
  getSearchResults: (query: string, categories: string[], refresh?: boolean) => void;
}

export const searchParams = (query: string, categories: string[]) => ({
  categories,
  query,
  queryId: getQueryId(query, categories)
});

export class Search extends React.Component<IProps> {

  static navigationOptions = (props: NavigationScreenProps) => ({
    headerRight: headerRight(props),
    title: props.navigation.getParam("query", "")
  });

  componentDidMount() {
    this.fetchResults();
    const [query, categories] = getNavigationParams({ query: "", categories: [] }, this.props.navigation);
    this.props.addRecentSearch(query, categories);
  }

  fetchResults = () => {
    const { navigation } = this.props;
    this.props.getSearchResults(navigation.getParam("query"), navigation.getParam("categories"));
  };

  renderItem = ({ item }: any) => (
    <RenderItem item={item} />
  );

  onPressSuggestion = (query: string) => {
    const [categories] = getNavigationParams({ categories: [] }, this.props.navigation);
    this.props.navigation.push(
      "Search",
      searchParams(query, categories)
    );
  };

  keyExtractor = (item: any, index: number) => `${item.item_id}-${index}`;

  renderSuggestions = () => {
    const { suggestions } = this.props;
    return (
      <View style={{ padding: 8 }}>
        <Text>You might be interested in: </Text>
        <View style={{ flexWrap: "wrap", flexDirection: "row", alignItems: "flex-start" }}>
          {
            suggestions.map((suggestion, index) => (
              <Suggestion
                key={`${suggestion}-${index}`}
                onPressSuggestion={this.onPressSuggestion}
                suggestion={suggestion}
              />
            ))
          }
        </View>
      </View>
    );
  };

  render() {
    const { results, isLoading } = this.props;
    return (
      <View>
        {
          (results.length || isLoading) ?
            <FlatList
              keyExtractor={this.keyExtractor}
              data={results}
              renderItem={this.renderItem}
              onEndReached={this.fetchResults}
              ListFooterComponent={<Loading isLoading={isLoading} />}
              ListHeaderComponent={this.props.suggestions.length ? this.renderSuggestions() : null}
            />
            :
            <Text>No results</Text>
        }
      </View>
    );
  }
}

const mapStateToProps = () => {
  const selectSearchResults = makeSelectSearchResults();
  return (state: IReducerState, props: NavigationScreenProps) => selectSearchResults(state, props);
};

const mapDispatchToProps = (dispatch: any) => ({
  addRecentSearch: (query: string, categories: string[]) => dispatch(recentSearches.addSearch(query, categories)),
  getSearchResults: (query: string, categories: string[], refresh = false) => {
    dispatch(getSearchResults(query, categories, refresh));
  }
});

const enhance = hoistStatics<IProps>(
  compose(
    connect(mapStateToProps, mapDispatchToProps),
    shouldUpdate(
      (props, nextProps: IProps) => nextProps.navigation.isFocused() && props.isLoading !== nextProps.isLoading
    )
  )
);

export default enhance(Search);
