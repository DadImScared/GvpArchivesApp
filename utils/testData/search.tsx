import * as React from "react";
import * as renderer from "react-test-renderer";

import { action } from "@storybook/addon-actions";
import _ from "lodash";

import { getQueryId } from "../../reducers/search";

export const makeTree = (Component: any, props: any = {}) => (injectedProps: any = {}) => {
  return renderer.create(<Component {...props} {...injectedProps} />);
};

const params = {
  categories: ["book", "song"],
  query: "radha",
  queryId: getQueryId("radha", ["book", "song"])
} as any;

export const getMockFunction = ({
  storybook = false,
  actionName = "",
  cb = false
}: any) => {
  if (storybook) {
    return (...args: any) => {
      action(actionName)(...args);
      if (cb) {
        return cb(...args);
      }
    };
  }
  return cb ? jest.fn(cb) : jest.fn();
};

export const searchTestProps = ({ props = {}, storybook = false } = {}) => (_.merge({
  addRecentSearch: getMockFunction({ storybook, actionName: "addRecentSearch" }),
  getSearchResults: getMockFunction({ storybook, actionName: "getSearchResults" }),
  isLoading: false,
  navigation: {
    getParam: getMockFunction({ storybook, actionName: "getParam", cb: (param: string) => params[param] }),
    push: getMockFunction({ storybook, actionName: "push" })
  },
  results: [
    {
      category: "book",
      item_id: "item-1",
      link: "item-1-link",
      title: "item-1-title"
    }
  ],
  suggestions: ["Suggestion-1", "Suggestion-2", "Suggestion-3"],
  timeToLive: 12345
}, props));
