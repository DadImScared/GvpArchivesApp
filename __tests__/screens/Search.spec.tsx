import * as React from "react";
import * as renderer from "react-test-renderer";

import { Button } from "native-base";

import { Search } from "../../screens/Search";
import { makeTree, searchTestProps } from "../../utils/testData/search";

export type TreeMaker = (injectedProps?: any) => renderer.ReactTestRenderer;

describe("Search", () => {
  let props: any;
  let treeFactory: TreeMaker;
  beforeEach(() => {
    props = searchTestProps();
    treeFactory = makeTree(Search, props);
  });

  it("should render with results", () => {
    expect(treeFactory().toJSON()).toMatchSnapshot();
  });

  it("should navigate on suggestion press", () => {
    const tree = treeFactory();
    const instance = tree.root;
    instance.findAllByType(Button)[0].props.onPress();
    expect(props.navigation.push.mock.calls[0]).toMatchSnapshot();
  });

  it("should get results on mount", () => {
    treeFactory();
    expect(props.getSearchResults.mock.calls[0]).toMatchSnapshot();
  });

  it("should add recent search on mount", () => {
    treeFactory();
    expect(props.addRecentSearch.mock.calls[0]).toMatchSnapshot();
  });
});
