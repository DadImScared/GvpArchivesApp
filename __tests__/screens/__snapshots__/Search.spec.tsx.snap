// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should add recent search on mount 1`] = `
Array [
  "radha",
  Array [
    "book",
    "song",
  ],
]
`;

exports[`Search should get results on mount 1`] = `
Array [
  "radha",
  Array [
    "book",
    "song",
  ],
]
`;

exports[`Search should navigate on suggestion press 1`] = `
Array [
  "Search",
  Object {
    "categories": Array [
      "book",
      "song",
    ],
    "query": "Suggestion-1",
    "queryId": "Suggestion-1-book,song",
  },
]
`;

exports[`Search should render with results 1`] = `
<View>
  <RCTScrollView
    ListFooterComponent={
      <Unknown
        isLoading={false}
      />
    }
    ListHeaderComponent={
      <Component
        style={
          Object {
            "padding": 8,
          }
        }
      >
        <Styled(Text)>
          You might be interested in: 
        </Styled(Text)>
        <Component
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "flexWrap": "wrap",
            }
          }
        >
          <Suggestion
            onPressSuggestion={[Function]}
            suggestion="Suggestion-1"
          />
          <Suggestion
            onPressSuggestion={[Function]}
            suggestion="Suggestion-2"
          />
          <Suggestion
            onPressSuggestion={[Function]}
            suggestion="Suggestion-3"
          />
        </Component>
      </Component>
    }
    data={
      Array [
        Object {
          "category": "book",
          "item_id": "item-1",
          "link": "item-1-link",
          "title": "item-1-title",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            uppercase={false}
          >
            You might be interested in: 
          </Text>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#007aff",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "space-between",
                  "margin": 8,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                  }
                }
                uppercase={false}
              >
                SUGGESTION-1
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#007aff",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "space-between",
                  "margin": 8,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                  }
                }
                uppercase={false}
              >
                SUGGESTION-2
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#007aff",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 5,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "space-between",
                  "margin": 8,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                  }
                }
                uppercase={false}
              >
                SUGGESTION-3
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "padding": 4,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              }
            }
          >
            <View
              header={true}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderBottomWidth": null,
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 15,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 15,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "stretch",
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontFamily": "System",
                          "fontSize": 16,
                        },
                        Object {
                          "width": "80%",
                        },
                      ]
                    }
                    uppercase={false}
                  >
                    item-1-title
                  </Text>
                  <Icon
                    name="book"
                    style={
                      Object {
                        "color": "#009688",
                      }
                    }
                    type="FontAwesome"
                  />
                </View>
                <Text
                  note={true}
                  style={
                    Object {
                      "color": "#808080",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "200",
                      "marginRight": 20,
                    }
                  }
                  uppercase={false}
                >
                  Books
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      />
    </View>
  </RCTScrollView>
</View>
`;
